### ISSUE TITLE
Purpose of Range?

### ISSUE THREAD

**Issue by maxwelljens** [cberner/redb#746]
I really like redb, but one part I am unable to figure out is what the purpose of `Range` is. As the README clearly states, and the documentation confirms, is that redb is a "key-value store". However, the `iter()` method on `Table` struct returns a `Range<'_, K, V>, StorageError`. If elements in a table are just keys with associated values, what is the goal of `Range`? I studied the documentation for a while but I was unable to figure out the purpose of it.

I did figure out how to work with it and iterate over the contents, but the loops are slightly more complicated than needed for a simple key-value table. It looks like I might be missing something.

**Comment by cberner** [comment]
It has the same purpose as `Range` in `std::BtreeMap`. It's the iterator itself

**Comment by maxwelljens** [comment]
Thank you for the quick response. I will make sure to research the topic further with the pointer.

**Comment by matthiasdebernardini** [comment]
I am also having issues with the Range trait

Here I am trying to take a table that has (&str, Vec<&str>) and I want to flatten the whole thing to a Vec<(&str, &str) but I don't understand how the Range struct works. 

I was hoping this would work but it doesn't.

```
    let read_txn = db.begin_read()?;
    let table = read_txn.open_table(TABLE)?;

    let mut result: Vec<(&str, &str)> = Vec::new();

    for (key, values) in table.range(..) {
        for value in values {
            result.push((key, value));
        }
    }
```

How would I turn the table into a vector of tuples?

Any help is appreciated.

**Comment by matthiasdebernardini** [comment]
I've managed to hack this, but I'm curious if there is a better way

```rust
    let read_txn = db.begin_read()?;
    let table = read_txn.open_table(TABLE)?;

    let mut temp: Vec<(&str, &str)> = Vec::new();

    while table.is_empty().unwrap().not() {
        let r  = table.first().unwrap().unwrap();
        let result: Vec<(&str, &str)> = r.1.iter().map(|&s| (r.0, s)).collect();
        result.iter().map(|tuple| temp.push(tuple.to_owned()))
    }
```

**Comment by cberner** [comment]
No, the values share the lifetime of the `AccessGuard`, so if you want `String`s you'll need to clone them like you're doing
